src/finch/compiled.py

```diff
@@ -29,6 +34,45 @@ def _to_lazy_tensor(x: Tensor | Any, /) -> Tensor | Any:
     return x if not isinstance(x, Tensor) else lazy(x)
 
 
+def _recurse_iter_return(x: IterObj, /, *, compute_kwargs: dict[str, Any]) -> IterObj:
+    current = 0
+
+    def recursor(_: Any) -> tuple[int, Any]:
+        nonlocal current
+        ret = current
+        current += 1
+        return ret
+
+    # Make a recursive iterator of indices.
+    idx_obj = _recurse(x, f=recursor)
+    jl_computed = []
+    py_computed = []
+
+    # Collect lazy tensors; use placeholder
+    _placeholder = object()
+    for x in _recurse_iter(x):
+        if isinstance(x, Tensor) and not x.is_computed():
+            jl_computed.append(x._obj)
+            py_computed.append(_placeholder)
+        else:
+            py_computed.append(x)
+
+    if len(jl_computed) == 1:
+        # This doesn't return an iterable of arrays -- only a single array
+        jl_computed = [jl.Finch.compute(*jl_computed, **compute_kwargs)]
+    else:
+        jl_computed = jl.Finch.compute(*jl_computed, **compute_kwargs)
```

Comment:
Nitpick: Just my preference, works either way, but how about:
```python
jl_len = len(jl_computed)
jl_computed = jl.Finch.compute(*jl_computed, **compute_kwargs)
if jl_len == 1:
    jl_computed = [jl_computed]
```

Reply:
Good suggestion! Done in https://github.com/finch-tensor/finch-tensor-python/pull/94/commits/61455d58946d2f5fe8858c39eb9e14b55b941adc.